src/lib/support/jsontlv/JsonToTlv.cpp

```diff
@@ -466,4 +466,25 @@ CHIP_ERROR ConvertTlvTag(const uint64_t tagNumber, TLV::Tag & tag)
 {
     return InternalConvertTlvTag(tagNumber, tag);
 }
+
+CHIP_ERROR JsonToTlvWithoutStruct(const std::string & jsonString, MutableByteSpan & tlv)
+{
+    Platform::ScopedMemoryBufferWithSize<uint8_t> buf;
+    VerifyOrReturnError(buf.Calloc(tlv.size()), CHIP_ERROR_NO_MEMORY);
+    MutableByteSpan dataWithStruct(buf.Get(), buf.AllocatedSize());
+    ReturnErrorOnFailure(JsonToTlv(jsonString, dataWithStruct));
+    TLV::TLVReader tlvReader;
+    TLV::TLVType outerContainer = TLV::kTLVType_Structure;
+    tlvReader.Init(dataWithStruct);
+    ReturnErrorOnFailure(tlvReader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+    ReturnErrorOnFailure(tlvReader.EnterContainer(outerContainer));
+    ReturnErrorOnFailure(tlvReader.Next());
+
+    TLV::TLVWriter tlvWrite;
+    tlvWrite.Init(tlv);
+    ReturnErrorOnFailure(tlvWrite.CopyElement(TLV::AnonymousTag(), tlvReader));
```

Comment:
The tag situation here is not documented in the API docs.  And why are we assuming the struct only has one element?

This seems like a _very_ special-cased API of some sort, that expects some _very_ specific input and does something weird with it....

Reply:
thanks. This acts the convenient function used for IM write. update the detailed comment